plugins/modules/opensearch.py

```diff
@@ -0,0 +1,1194 @@
+        type: list
+        elements: dict
+        default: []
+    type: dict
+  wait:
+    description:
+      - Whether or not to wait for snapshot creation or deletion.
+    type: bool
+    default: 'no'
+  wait_timeout:
+    description:
+      - how long before wait gives up, in seconds.
+    default: 300
+    type: int
+  tags:
+    description:
+      - tags dict to apply to a snapshot.
+    type: dict
+  purge_tags:
+    description:
+      - whether to remove tags not present in the C(tags) parameter.
+    default: True
+    type: bool
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+"""
+
+EXAMPLES = """
+
+- name: Create OpenSearch domain
+  community.aws.opensearch:
+    domain_name: "my-domain"
+    engine_version: OpenSearch_1.1
+    cluster_config:
+      instance_type: "m3.medium.search"
+      instance_count: 2
+      dedicated_master: true
+      zone_awareness: true
+      availability_zone_count: 2
+      dedicated_master_instance_type: "t2.micro.search"
+      dedicated_master_instance_count: 2
+      warm_enabled: true
+      warm_type: "ultrawarm1.medium.search"
+      warm_count: 1
+      cold_storage_options:
+        enabled: false
+    ebs_options:
+      enabled: true
+      volume_type: "io1"
+      volume_size: 10
+      iops: 1000
+    vpc_options:
+      subnets:
+        - "subnet-e537d64a"
+        - "subnet-e537d64b"
+      security_groups:
+        - "sg-dd2f13cb"
+        - "sg-dd2f13cc"
+    snapshot_options:
+      automated_snapshot_start_hour: 13
+    access_policies: "{{ lookup('file', 'policy.json') | from_json }}"
+    encryption_at_rest_options:
+      enabled: false
+    node_to_node_encryption_options:
+      enabled: false
+    auto_tune_options:
+      enabled: true
+      maintenance_schedules:
+      - start_at: "2025-01-12"
+        duration:
+          value: 1
+          unit: "HOURS"
+        cron_expression_for_recurrence: "cron(0 12 * * ? *)"
+      - start_at: "2032-01-12"
+        duration:
+          value: 2
+          unit: "HOURS"
+        cron_expression_for_recurrence: "cron(0 12 * * ? *)"
+    tags:
+      Environment: Development
+      Application: Search
+    wait: true
+
+"""
+
+from copy import deepcopy
+import json
+import time
+
+try:
+    import botocore
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+from ansible.module_utils.six import string_types
+
+# import module snippets
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import (
+    camel_dict_to_snake_dict,
+    AWSRetry,
+    compare_aws_tags,
+)
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import (
+    boto3_tag_list_to_ansible_dict,
+    ansible_dict_to_boto3_tag_list,
+)
+
+
+def get_domain(client, module, domain_name):
+    try:
+        response = client.describe_domain(DomainName=domain_name)
+    except client.exceptions.ResourceNotFoundException:
```

Comment:
Import `from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code`

You can use something like this instead (even for the other cases)

```suggestion
    except is_boto3_error_code('ResourceNotFoundException'):
```